### PR TITLE
read hint

### PR DESCRIPTION
**Pull request by msmouse**
### Description
Real(er) read hints for p2p txns. 

### Test Plan

Printing out all fall back reads from the ptx executer and now seeing None data reads (module reads are still falling back to the base view)

### CODE REVIEW

**Review by grao1991 — APPROVED** [review]

File: types/src/transaction/analyzed_transaction.rs [hunk 1]
```diff
@@ -131,11 +133,18 @@ impl AnalyzedTransaction {
             // receiver account resource.
             write_hints.push(Self::account_resource_location(receiver_address));
         }
+
+        let read_hints = vec![
```
Comment (sitalkedia) [inline types/src/transaction/analyzed_transaction.rs:137]: Should we also add the framework code location here?

File: types/src/transaction/analyzed_transaction.rs [hunk 2]
```diff
@@ -131,11 +133,18 @@ impl AnalyzedTransaction {
...
+            Self::current_ts_location(),
+            Self::features_location(),
+            Self::aptos_coin_info_location(),
+            Self::chain_id_location(),
+            Self::transaction_fee_burn_cap_location(),
```
Comment (sitalkedia) [inline types/src/transaction/analyzed_transaction.rs:142]: Also, how about account creation txn?

File: types/src/transaction/analyzed_transaction.rs (hunk 1 above)
Comment (msmouse) [reply types/src/transaction/analyzed_transaction.rs:137]: That's a lot of modules, I'm thinking maybe we need to come up with some clever tricks down the road. For example, assuming commonly needed modules for all shards.

File: types/src/transaction/analyzed_transaction.rs (hunk 2 above)
Comment (msmouse) [reply types/src/transaction/analyzed_transaction.rs:142]: Will need to print some logs to figure out and confirm. I'd test the performance impact with p2p first.

**Review by sitalkedia — APPROVED** [review]

File: types/src/transaction/analyzed_transaction.rs (hunk 2 above)
Comment (msmouse) [reply types/src/transaction/analyzed_transaction.rs:142]: fails your test already :trollface: 
test_no_conflict_across_shards_in_non_last_rounds

Comment (sitalkedia) [reply types/src/transaction/analyzed_transaction.rs:142]: Let's rebase once https://github.com/aptos-labs/aptos-core/pull/9883 lands. I moved all these tests from v1 to v2. We will see if they fail in V2. cc - @zjma

**Pull request merged**